source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureBase.cs

```diff
@@ -89,44 +84,81 @@ CalamariResult ExecuteDirectlyWithScriptEngine(IReadOnlyList<IScriptWrapper> wra
             return new CalamariResult(result.ExitCode, new CaptureCommandInvocationOutputSink());
         }
 
-        protected void DeployWithScriptAndVerifySuccess(Action<TemporaryDirectory> addFilesAction = null)
+        protected void DeployWithDeploymentScriptAndVerifyResult(Func<string, string> addFilesOrPackageFunc = null, bool shouldSucceed = true)
         {
-            SetupTempDirectoryAndVerifyResult(addFilesAction, d =>
+            DeployWithScriptAndVerifyResult(() =>
             {
                 var scriptPath = Path.Combine(testFolder, "KubernetesFixtures/Scripts");
+                var bashScript = File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.sh"));
+                var powershellScript = File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.ps1"));
+                return (bashScript, powershellScript);
+            }, addFilesOrPackageFunc, shouldSucceed);
+        }
+
+        protected void DeployWithKubectlTestScriptAndVerifyResult()
+        {
+            DeployWithScriptAndVerifyResult("#{Octopus.Action.Kubernetes.CustomKubectlExecutable} cluster-info");
+        }
+
+        protected void DeployWithNonKubectlTestScriptAndVerifyResult()
+        {
+            DeployWithScriptAndVerifyResult("echo running target script...");
+        }
+
+        private void DeployWithScriptAndVerifyResult(string script,
+            Func<string, string> addFilesOrPackageFunc = null, bool shouldSucceed = true)
+        {
+            DeployWithScriptAndVerifyResult(() => (script, script), addFilesOrPackageFunc, shouldSucceed);
+        }
+
+        private void DeployWithScriptAndVerifyResult(Func<(string bashScript, string powershellScript)> scriptFactory,
+            Func<string, string> addFilesOrPackageFunc = null, bool shouldSucceed = true)
+        {
+            SetupTempDirectoryAndVerifyResult(addFilesOrPackageFunc, (d, p) =>
+            {
+                var (bashScript, powershellScript) = scriptFactory();
                 variables.Set(SpecialVariables.Action.Script.ScriptBodyBySyntax(ScriptSyntax.Bash),
-                    File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.sh")));
+                    bashScript);
                 variables.Set(SpecialVariables.Action.Script.ScriptBodyBySyntax(ScriptSyntax.PowerShell),
-                    File.ReadAllText(Path.Combine(scriptPath, "KubernetesDeployment.ps1")));
+                    powershellScript);
```

Comment:
Can we just set the variables within the `DeployWithScriptAndVerifyResult` method instead of doing it within the  `SetupTempDirectoryAndVerifyResult` method?

Reply:
Fixed (see other comment)